lib/widgets/branch_page.dart

```diff
@@ -0,0 +1,163 @@
+  String _title = 'Учеба';
+
+  bool _isCompletedFilter = false;
+  bool _isFavoriteFilter = false;
+
+  final List<Task> _tasks = [];
+  List<Task> _filteredTasks = [];
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: const Color(0xffb5c9fd),
+      appBar: createAppBar(),
+      body: _filteredTasks.isEmpty ? const EmptyBranch() : createListView(),
+      floatingActionButton: FloatingActionButton(
+        onPressed: _showAddTaskDialog,
+        tooltip: 'Add task',
+        backgroundColor: Colors.teal,
+        child: const Icon(Icons.add),
+      ),
+    );
+  }
+
+  Future<void> _showAddTaskDialog() async {
+    final enteredText = await showDialog<String>(
+      context: context,
+      builder: (_) => const TaskCreationDialog(),
+    );
+
+    if (enteredText != null) {
+      if (mounted) {
+        _tasks.add(
+          Task(
+            title: enteredText,
+            id: const Uuid().v4(),
+          ),
+        );
+        _filter();
+      }
+    }
+  }
+
+  Future<void> _showEditBranchDialog() async {
+    final enteredText = await showDialog<String>(
+      context: context,
+      builder: (_) => RenameBranchDialog(branchTitle: _title),
+    );
+
+    if (enteredText != null) {
+      if (mounted) {
+        setState(() {
+          _title = enteredText;
+        });
+      }
+    }
+  }
+
+  Future<void> _showDeleteDialog() async {
+    await showDialog<String>(
+      context: context,
+      builder: (_) => ConfirmationDialog(
+        onOk: _deleteCompletedTasks,
+        title: 'Подтвердите удаление',
+      ),
+    );
+  }
+
+  void _toggleCompletion({required String taskId}) {
+    final index = _tasks.indexWhere((element) => element.id == taskId);
+
+    _tasks[index] = _tasks[index].copyWith(isCompleted: !_tasks[index].isCompleted);
+    _filter();
+  }
+
+  void _toggleFavourite({required String taskId}) {
+    final index = _tasks.indexWhere((element) => element.id == taskId);
+    _tasks[index] = _tasks[index].copyWith(isFavourite: !_tasks[index].isFavourite);
+
+    _filter();
+  }
+
+  void _deleteTask({required String taskId}) {
+    _tasks.removeWhere((element) => element.id == taskId);
+    _filter();
+  }
+
+  void _deleteCompletedTasks() {
+    _tasks.removeWhere((task) => task.isCompleted);
+    _filter();
+  }
+
+  void _onFavoriteFilter() {
+    _isFavoriteFilter = !_isFavoriteFilter;
+    _filter();
+  }
+
+  void _onCompletedFilter() {
+    _isCompletedFilter = !_isCompletedFilter;
+    _filter();
+  }
+
+  void _filter() {
+    Iterable<Task> showTasksIterable = _tasks;
+    if (_isCompletedFilter) {
+      showTasksIterable = showTasksIterable.where((task) => !task.isCompleted);
+    }
+    if (_isFavoriteFilter) {
+      showTasksIterable = showTasksIterable.where((task) => task.isFavourite);
+    }
+    setState(() {
+      _filteredTasks = showTasksIterable.toList();
+    });
+  }
+
+  ListView createListView() => ListView.separated(
```

Comment:
```suggestion
  Widget createListView() => ListView.separated(
```
Лучше не конкретизировать без необходимости